controllers/mysqlserver_controller.go

```diff
@@ -24,7 +33,66 @@ func (r *MySQLServerReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 }
 
 func (r *MySQLServerReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	// In order to watch a secret associated with this server:
+	// * We cannot use "For" - this automatically reconciles the type in question, and since the type
+	//   is a secret we don't want that.
+	// * We cannot use "Owns" - this reconciles the resources owner.
+	// * We must use Watches
+
+	// TODO:
+	// In order to actually make a change to the MySQLServer password, you must issue an update to the server which
+	// includes "administratorLogin": "<user>", "administratorLoginPassword": "<password>",
+
+	// Add a field indexer which we will use to find servers
+	err := mgr.GetFieldIndexer().IndexField(context.Background(), &v1alpha2.MySQLServer{}, AdminSecretKey, func(rawObj client.Object) []string {
+		obj := rawObj.(*v1alpha2.MySQLServer)
+		return []string{obj.Spec.AdminSecret}
+	})
+	if err != nil {
+		return err
+	}
+
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&v1alpha2.MySQLServer{}).
+		Watches(&source.Kind{Type: &corev1.Secret{}}, handler.EnqueueRequestsFromMapFunc(makeMapFunc(mgr))).
```

Comment:
This is neat!

Reply:
Yeah, it's a bit awkward right now because we end up watching all secrets. I'm not 100% sure we can avoid that, we might be able to with https://github.com/kubernetes-sigs/controller-runtime/blob/master/designs/use-selectors-at-cache.md but I couldn't work out how to do it and it's an optimization not required for functionality so I've left a note but we should be ok for now I think.